### PR TITLE
Replace bundle-loader with react-universal-component, refactor Union and scanning for SSR

### PR DESCRIPTION
**Pull request by wafflepie**
I did not have time to refactor a test in `Union-test`, so I commented it out. Tests and documentation are missing as of yet.

### CODE REVIEW

File: packages/react-union/src/components/Union/Union.js [hunk 1]
```diff
@@ -48,41 +60,53 @@ class Union extends Component {
...
+		this.setState({
+			configs: this.scan(this.props),
+		});
 	}
 
 	componentDidUpdate(prevProps) {
 		if (prevProps.routes !== this.props.routes) {
-			this.scan(prevProps);
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState({
```
Comment (aizerin) [inline packages/react-union/src/components/Union/Union.js]: I know that this is a WIP, but you can use `getDerivedStateFromProps` here.

**Comment by wafflepie** [comment]
I've fixed some of the issues with the tests and Union itself (as well as removed the WidgetProvider component, because it was unnecessary). Documentation is still TBD.

**Comment by wafflepie** [comment]
I'm trying to integrate react-loadable into my SSR server and I'm having issues with hot reloading. I will try some other libraries as well (I've got my eye on react-universal-component).

**Comment by wafflepie** [comment]
We will create a new branch for SSR, closing this for now.

**Pull request closed without merging**